src/VerticalGrids.jl

```diff
@@ -23,75 +26,161 @@ function (stretching::LinearStretching)(Δz, z)
+                        rounding_digits = 2)
+
+Return a type that describes a one-dimensional grid with `surface_layer_Δz` spacing
+in a surface layer of extent `surface_layer_height`, and stretched according to
+the `stretching` down to `depth`.
+The interfaces extend from `depth = -z[1]` to `0 = z[end]`, where `Lz ≥ depth`.
+
+The grid spacing `Δz` is limited to be less than `maximum_Δz`.
+The grid is also uniformly-spaced below `constant_bottom_spacing_depth`.
 
-    return z
+`rounding_digits` controls the accuracy with which the grid interface positions are saved.
+"""
+function StretchedInterfaces(; depth = 5000,
+                             surface_layer_Δz = 5.0,
+                             surface_layer_height = 100.0,
+                             constant_bottom_spacing_depth = Inf,
+                             maximum_Δz = Inf,
+                             stretching = PowerLawStretching(1.02),
+                             rounding_digits = 2)
+
+    return StretchedInterfaces(depth,
+                               surface_layer_Δz,
+                               surface_layer_height,
+                               constant_bottom_spacing_depth,
+                               maximum_Δz,
+                               stretching;
+                               rounding_digits)
 end
 
-@inline exponential_profile(z, Lz, h) = (exp(z / h) - exp(-Lz / h)) / (1 - exp(-Lz / h))
+(g::StretchedInterfaces)(k) = @inbounds g.faces[k]
+
+Base.length(g::StretchedInterfaces) = length(g.faces)-1
+
+@inline exponential_profile(z, L, h) = @. - L * expm1(- z/ h) / expm1(L / h)
 
-function exponential_z_faces(; Nz, depth, h = Nz / 4.5)
+struct ExponentialInterfaces{FT} <: Function
+    size :: Int
+    extent :: FT
+    scale :: FT
 
-    k = collect(1:Nz+1)
-    z_faces = exponential_profile.(k, Nz, h)
+    @doc """
+        ExponentialInterfaces(size::Int, extent; scale=extent/5)
 
-    # Normalize
-    z_faces .-= z_faces[1]
-    z_faces .*= - depth / z_faces[end]
+    Return a type that describes a one-dimensional coordinate with `N+1` faces (i.e., `N` cells) that
+    are exponentially spaced (or, equivalently, with spacings that grow linearly with depth)
+    with `extent`. The coordinate spans `[-depth, 0]`. The exponential scaling is controlled by
+    keyword argument `scale` (default: `extent/5`).
+    """
+    function ExponentialInterfaces(size::Int, extent; scale=extent/5)
+        FT = typeof(scale)
+        return new{FT}(size, extent, scale)
+    end
+end
 
-    z_faces[1] = 0.0
+Base.summary(g::ExponentialInterfaces) = "ExponentialInterfaces"
+
+function Base.show(io::IO, g::ExponentialInterfaces)
+    print(io, summary(g), '\n')
+    print(io, "├── size: ", prettysummary(g.size), '\n')
+    print(io, "├── extent: ", prettysummary(g.extent), '\n')
+    print(io, "└── scale: ", prettysummary(g.scale), '\n')
+end
+
+function (g::ExponentialInterfaces)(k)
+    Nz, depth, scale = g.size, g.extent, g.scale
+
```

Comment:
I'm wondering whether a validation is appropriate here, e.g.,

```suggestion

(!isinteger(k) || k < 1 || k > Nz+1) && throw(ArgumentError("only integer arguments within 1 and $(Nz+1)"))

```